### PR TITLE
H-69: Update the "archived page" banner

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR updates the "archive page" banner to display the person who archived the page, alongside the timestamp.

## 🔍 What does this change?

- fixes restoring pages, which was previously broken because it wasn't possible to set properties to `false` via the `updateEntityProperties` method in the TS API

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## 🛡 What tests cover this?

None

## ❓ How to test this?

1. Login and archive a page in the sidebar
2. View the updated content of the archived page banner
3. Use the "restore" button to unarchive the page
4. Ensure the "Page Restored!" message is displayed

## 📹 Demo

https://github.com/hashintel/hash/assets/42802102/ce81dea2-31f4-4dca-a282-a32c7aadee84

### CODE REVIEW

**Comment by nonparibus** [comment]
@benwerner01 I realize this is a WIP, but a few initial observations (which I figure you're probably already aware of):
- when a page is archived, the breadcrumbs/its name cease to appear.
- when a page is unarchived, the "this page has been archived" banner persists, even after the "page unarchived" success message shows. Upon refresh the banner is still there.

Tested with canvases, but I believe also occurring with documents. Sorry for jumping the gun to have a play with this... I was excited! No need to reply here.

**Comment by benwerner01** [comment]
> when a page is unarchived, the "this page has been archived" banner persists, even after the "page unarchived" success message shows. Upon refresh the banner is still there.

@nonparibus if you're testing this in the deployment, that'll be because the BE has a bug (which this PR fixes) which causes the functionality of restoring pages to break. Since we aren't deploying the BE to a staging environment on a per PR basis, the functionality can only be tested locally atm.

Will take a look at why the breadcrumbs are disappearing.

**Comment by nonparibus** [comment]
@benwerner01 Thanks. The green color of the success chip deviates from that used in the designs, but this is because:
- Our old `Green` was renamed to `Lime` in the design system.
- Our new `Green` is our old `Mint`.

We can update these globally separately. I've captured this [as a follow-up task]((https://linear.app/hash/issue/H-352)) in Linear (internal link).

File: libs/@local/hash-graphql-shared/src/graphql/type-defs/knowledge/page.typedef.ts [hunk 1]
```diff
@@ -68,6 +68,10 @@ export const pageTypedef = gql`
       The account owning the pages. Defaults to the logged in user.
       """
       ownedById: OwnedById
+      """
+      Whether or not to include archived pages. Defaults to false.
+      """
+      includeArchived: Boolean
```
Comment (CiaranMn) [inline libs/@local/hash-graphql-shared/src/graphql/type-defs/knowledge/page.typedef.ts:74]: We can express the default in the definition like so. 

Unfortunately the coden doesn't understand that there's a default so the types will still have it as undefined, but the actual value passed to the function will be `false` (codegen may have been updated to handle this, haven't checked in a while)
```suggestion
      includeArchived: Boolean = false
```

File: apps/hash-frontend/src/blocks/page/page-section-container.tsx [hunk 2]
```diff
@@ -8,11 +8,11 @@ export const PAGE_MIN_PADDING = 48;
 export const COMMENTS_WIDTH = 320;
 
 export const getPageSectionContainerStyles = (
-  pageComments: PageThread[],
+  pageComments?: PageThread[],
   readonlyMode?: boolean,
 ) => {
   const commentsContainerWidth =
-    !readonlyMode && pageComments.length
+    !readonlyMode && pageComments && pageComments.length
```
Comment (CiaranMn) [inline apps/hash-frontend/src/blocks/page/page-section-container.tsx:15]:
```suggestion
    !readonlyMode && pageComments?.length
```

File: apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx [hunk 3]
```diff
@@ -186,47 +194,168 @@ export const getServerSideProps: GetServerSideProps<PageProps> = async ({
...
+  const { unarchivePage } = useArchivePage();
+
+  const { provenance, temporalVersioning } = pageMetadata;
+
+  const { users } = useUsers();
+
+  const archivedByAccountId = provenance.recordCreatedById;
+
+  const archivedByUser =
+    users && users.find(({ accountId }) => archivedByAccountId === accountId);
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx:214]:
```suggestion
  const archivedByUser = users?.find(({ accountId }) => archivedByAccountId === accountId);
```

File: apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx [hunk 4]
```diff
@@ -186,47 +194,168 @@ export const getServerSideProps: GetServerSideProps<PageProps> = async ({
...
+        return `${differenceInHours} hours`;
+      } else {
+        const differenceInMinutes = Math.round(
+          differenceInMilliseconds / (1000 * 60),
+        );
+
+        return `${differenceInMinutes} minutes`;
+      }
+    }
+  }, [archivedAt]);
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/[shortname]/[page-slug].page.tsx:246]: We have `date-fns` in the repo [[docs](https://date-fns.org/v2.30.0/docs/formatDistance)]
```suggestion
  const timeSinceArchived = formatDistance(archivedAt, new Date());
```

**Review by CiaranMn — APPROVED** [review]
Looks good! A few non-blocking comments

File: apps/hash-api/src/graph/knowledge/system-types/page.ts [hunk 5]
```diff
@@ -270,13 +272,13 @@ export const getAllPagesInWorkspace: ImpureGraphFunction<
     .filter(
```
[the remark this answers is not in the document]
Comment (benwerner01) [reply apps/hash-api/src/graph/knowledge/system-types/page.ts]: Addressed in https://github.com/hashintel/hash/pull/2876/commits/e067a00f68ecf5ef2efd281e6157ca6ed532b316

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**